pkg/api/v1/types.go

```diff
@@ -710,6 +725,22 @@ type Container struct {
 	TTY   bool `json:"tty,omitempty" description:"Whether this container should allocate a TTY for itself, also requires 'stdin' to be true; default is false"`
 }
 
+func (ctr *Container) ApplyDefaults() {
```

Comment:
To be clear:

validation in code is great and appropriate, tags are probably not a good fit for the reasons you mentioned.

defaulting in tags seems to make sense, it's self-documenting, and I don't think you should ever need code to do it.

--brendan